packages/plugins/jwt/src/utils.ts

```diff
@@ -0,0 +1,92 @@
+import { createGraphQLError } from 'graphql-yoga';
+import { JwksClient, type Options as JwksClientOptions } from 'jwks-rsa';
+import { GetSigningKeyFunction, type ExtractTokenFunction } from './config.js';
+
+export function extractFromHeader(options: {
+  name: string;
+  prefix?: string;
+}): ExtractTokenFunction {
+  return ({ request }) => {
+    const header = request.headers.get(options.name);
+
+    if (header == null) {
+      return;
+    }
+
+    if (!options.prefix) {
+      return header;
+    }
+
+    const [prefix, token] = header.split(' ').map(s => s.trim());
+
+    if (prefix !== options.prefix) {
+      throw badRequestError(`Invalid JWT authentication token prefix.`);
+    }
```

Comment:
I'm moving my comment there :-)

So to better explain myself, are prefixes standard ? Because in my previous job, I used to see pretty random stuff there ^^'

My idea is that one should be able to provide any prefix, without assumptions over it's content or the seperator used.

For example, if my header looks like `auth.xxx...xxxxx`, I would like to configure my prefix for being `auth.`.
If my token looks like `AUTH JWT xxxxxxx`, I would like to configure my prefix as `AUTH JWT` or ``AUTH JWT ` to be more strict about the seperator.

With this implementation, it always check the prefix, whatever this prefix is :-) 
```suggestion
    const [prefix, token] = header.split(options.prefix).map(s => s.trim());

    // The first part should be empty string when the string starts with the prefix
    if (prefix !== '') {
      throw badRequestError(`Invalid JWT authentication token prefix.`);
    }
```

Reply:
https://datatracker.ietf.org/doc/html/rfc1945#section-10.2 
https://stackoverflow.com/questions/68869644/what-is-the-reason-for-adding-bearer-as-the-prefix-to-your-web-token

The prefix part is part of the standard, but it's optional. I see your point, will improve the code now to support any prefix when it's not configured to validate.

Reply:
Fixed :)